src/core/operations/Typex.mjs

```diff
@@ -0,0 +1,398 @@
+/**
+ * Emulation of the Typex machine.
+ *
+ * @author s2224834
+ * @copyright Crown Copyright 2019
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation";
+import OperationError from "../errors/OperationError";
+import Utils from "../Utils";
+import * as Enigma from "../lib/Enigma";
+
+const ROTORS = [
+    {name: "1", value: "QWECYJIBFKMLTVZPOHUDGNRSXA<ACEINQTVY"},
+    {name: "2", value: "AJDKSIRUXBLHWTMCQGZNPYFVOE<ACEINQTVY"},
+    {name: "3", value: "BDFHJLCPRTXVZNYEIWGAKMUSQO<ACEINQTVY"},
+    {name: "4", value: "ESOVPZJAYQUIRHXLNFTGKDCMWB<ACEINQTVY"},
+    {name: "5", value: "VZBRGITYUPSDNHLXAWMJQOFECK<ACEINQTVY"},
+    {name: "6", value: "FVPJIAOYEDRZXWGCTKUQSBNMHL<ACEINQTVY"},
+    {name: "7", value: "KZGLIUCJEHADXRYWVTNSFQPMOB<ACEINQTVY"},
+    {name: "8", value: "ZLVGOIFTYWUEPMABNCXRQSDKHJ<ACEINQTVY"},
+];
+
+const REFLECTORS = [
+    {name: "Standard", value: "MO VW GL JX TZ AY EQ IP KN DH CU FS BR"},
+];
+
+// Special character handling on Typex keyboard
+const KEYBOARD = {
+    "Q": "1", "W": "2", "E": "3", "R": "4", "T": "5", "Y": "6", "U": "7", "I": "8", "O": "9", "P": "0",
+    "A": "-", "S": "/", "D": "Z", "F": "%", "G": "X", "H": "£", "K": "(", "L": ")",
+    "C": "V", "B": "'", "N": ",", "M": "."
+};
+const KEYBOARD_REV = {};
+for (const i of Object.keys(KEYBOARD)) {
+    KEYBOARD_REV[KEYBOARD[i]] = i;
+}
+
+/**
+ * Typex machine. A lot like the Enigma, but five rotors, of which the first two are static.
+ */
+class TypexMachine extends Enigma.EnigmaBase {
```

Comment:
I think I would move the TypexMachine, Rotor and Plugboard classes into src/core/lib/Typex.mjs. It will make it easier for other people/operations to use. It will also mean you can rename `TypexOp` to `Typex`. There is some logic in various configuration scripts that expects the operation class and filename to be the same, so this could mess things up a bit.

The code in src/core/operations should really only be concerned with making it work for CyberChef (although there are still lots of operations where this is not the case yet, it's something I'd like to work towards).